modules/standard/GMP.chpl

```diff
@@ -581,838 +602,993 @@ module GMP {
     proc get_ui():uint
     {
       var x:c_ulong;
-      on this do x = mpz_get_ui(this.mpz);
+      on Locales[this.locale_id] do x = mpz_get_ui(this.mpz);
       return x.safeCast(uint);
     }
     proc get_si():int
     {
       var x:c_long;
-      on this do x = mpz_get_si(this.mpz);
+      on Locales[this.locale_id] do x = mpz_get_si(this.mpz);
       return x.safeCast(int);
     }
     proc get_d():real
     {
       var x:c_double;
-      on this do x = mpz_get_d(this.mpz);
+      on Locales[this.locale_id] do x = mpz_get_d(this.mpz);
       return x:real;
     }
+
     // returns (exponent, double)
-    proc get_d_2exp():(int, real)
+    // TODO: compiler chooses an INT(64) for tmp, gets win-pointer mismatch
+    // for architectures where long and long long are both 8 bytes, it makes
+    // an int64 * for the long, and tries to return it as a long long * instead
+    // makes a warning
+    proc get_d_2exp():(uint(32), real)
     {
       var exp:c_long;
       var dbl:c_double;
-      on this {
+      on Locales[this.locale_id] {
         var tmp:c_long;
         dbl = mpz_get_d_2exp(tmp, this.mpz);
         exp = tmp;
       }
-      return (exp.safeCast(int), dbl: real);
+      return (exp.safeCast(uint(32)), dbl: real);
     }
-    proc get_str(base:int=10):string
+    proc get_str(base: int=10):string
     {
       var ret:string;
-      on this {
-        var tmp = chpl_gmp_mpz_get_str(base.safeCast(c_int), this.mpz);
-        ret = tmp:string;
+      on Locales[this.locale_id] {
+        var tmpvar = chpl_gmp_mpz_get_str(base.safeCast(c_int), this.mpz);
+        ret = new string(tmpvar, owned=true, needToCopy=false);
       }
       return ret;
     }
 
     // Arithmetic functions
-    proc add(a:BigInt, b:BigInt)
+    proc add(ref a: BigInt, ref b: BigInt)
```

Comment:
In response to 1):
Yes. They should be const ref. I just also have to change the extern declaration to permit that. 

In response to 2): 
How I have the named function interface set up is that for big ints a, b, and c, doing a.someFunction(b, c) should set a to the result of using that function on b and c.